### PR TITLE
Finish codegen references for queries and spts

### PR DESCRIPTION
**Pull request by ericanderson**
Chains to #580

### CODE REVIEW

File: packages/generator/src/GenerateContext/EnhancedBase.ts [hunk 1]
```diff
@@ -55,9 +55,7 @@ export abstract class AbstractImportable {
     this.importPath = this.isLocal
       ? `${basePath}/${this.shortApiName}${importExt}`
       : this.sourcePackage!;
-    this.uniqueImportName = this.isLocal
-      ? this.shortApiName
-      : `$external$${this.fullApiName.replace(".", "$")}`;
+    this.uniqueImportName = this.shortApiName;
```
Comment (ssanjay1) [inline packages/generator/src/GenerateContext/EnhancedBase.ts:58]: Why no need for the local check anymore?

File: packages/generator/src/v2.0/generatePerQueryDataFiles.ts [hunk 2]
```diff
@@ -193,22 +193,32 @@ function parametersForConst(query: EnhancedQuery) {
...
+) {
   if (qdt.type === "object" || qdt.type === "objectSet") {
-    return `__OsdkTargetType?: ${
-      getObjectDefIdentifier(qdt.objectTypeApiName!, true)
+    return `__OsdkTargetType?: ${ontology.requireObjectType(
+        qdt.objectTypeApiName!,
+      ).getImportedDefinitionIdentifier(
+        true,
+      )
+      // getObjectDefIdentifier(qdt.objectTypeApiName!, true)
```
Comment (ssanjay1) [inline packages/generator/src/v2.0/generatePerQueryDataFiles.ts:221]: Need this?

**Review by ssanjay1 — APPROVED** [review]
Approved with one question and one nit comment

File: packages/generator/src/GenerateContext/EnhancedBase.ts (hunk 1 above)
Comment (ericanderson) [reply packages/generator/src/GenerateContext/EnhancedBase.ts:58]: The functions I put on the object handle this now.

**Pull request merged**